tools/cli/internal/cli/root/openapi/builder.go

```diff
@@ -0,0 +1,66 @@
+// Copyright 2024 MongoDB Inc
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package openapi
+
+import (
+	"fmt"
+	"mongodb/openapi/tools/cli/internal/cli/merge"
+	"mongodb/openapi/tools/cli/internal/version"
+	"runtime"
+
+	"github.com/spf13/cobra"
+)
+
+const (
+	ToolName    = "openapicli"
+	verTemplate = `openapicli version: %s
+git version: %s
+Go version: %s
+   os: %s
+   arch: %s
+   compiler: %s
+`
+)
+
+// Builder conditionally adds children commands as needed.
+// This is important in particular for Atlas as it dynamically sets flags for cluster creation and
+// this can be slow to timeout on environments with limited internet access (Ops Manager).
+func Builder(argsWithoutProg []string) *cobra.Command {
```

Comment:
unused? maybe a good idea to add golangci-lint

Reply:
this is used in `openapicli.go`. Yes, I will add golanci-lint in a follow up PR

Reply:
either same as you have here first is base or we can have two flags `--base` `--other...` and why flags are usually more friendly

Reply:
but I don't see it used, this feels like a copy/past from atlas cli where we used to know if to enforce auth or not

Reply:
sorry to be clear the unused is `argsWithoutProg`

Reply:
ah sorry, yes I will remove it